sentry_sdk/tracing_utils.py

```diff
@@ -421,6 +421,7 @@ def __init__(
         parent_span_id=None,  # type: Optional[str]
         parent_sampled=None,  # type: Optional[bool]
         dynamic_sampling_context=None,  # type: Optional[Dict[str, str]]
```

Comment:
I'm not removing the constructor parameter because that would be technically breaking but it does nothing now. 
@sentrivana if we consider this class purely internal, I can also remove, wdyt?
Also, we never call this constructor with any arguments ourselves.